server/lib/watcher.js

```diff
@@ -0,0 +1,182 @@
+      logger.info("##### STARTING PLASMA CHECK #####");
+      logger.info("");
+      logger.info(await this.validateLeapPlasma());
+      logger.info("##### FINISHED PLASMA CHECK #####");
+      poller.poll(config.pollingDelay);
+    });
+
+    if (leap.web3) {
+      poller.poll();
+    } else {
+      setTimeout(this.init, 500);
+    }
+  }
+
+  async getUnspentTransactionSum() {
+    const { bridge, exitHandler, plasma, web3 } = leap;
+    const exits = [];
+
+    const uTxos = await plasma.getUnspentAll();
+    logger.info(`UNSPENT TRANSACTIONS: ${uTxos.length}`);
+
+    const addresses = await plasma.getColors();
+    logger.info(`ADDRESSES: ${addresses.length}`);
+
+    const colors = await Promise.all(
+      addresses.map(async address => {
+        return plasma.getColor(address);
+      })
+    );
+    logger.info(`COLORS: ${colors.length}`);
+
+    const uTxoSums = await Promise.all(
+      colors.map(async color => {
+        return uTxos.reduce((hexValue, uTxo) => {
+          if (uTxo.output.color === color) {
+            const value = new BN(uTxo.output.value).add(hexValue);
+
+            return value;
+          } else {
+            return hexValue;
+          }
+        }, new BN(0));
+      })
+    );
+
+    let utxs = 0;
+
+    for (let i = 0; i < uTxoSums.length; i++) {
+      const numberAmount = parseFloat(
+        web3.utils.fromWei(uTxoSums[i].toString(), "ether")
+      );
+
+      utxs += numberAmount;
+    }
+
+    this.state.uTxoSum = utxs;
+    logger.info(`PLASMA UTXO SUM: ${this.state.uTxoSum}`);
+  }
+
+  async getExitSum() {
+    const { bridge, exitHandler, web3, plasma } = leap;
+
+    const fromBlock = await bridge.methods.genesisBlockNumber().call();
+    const events = await exitHandler.getPastEvents("ExitStarted", {
+      fromBlock
+    });
+    let exits = [];
+
+    await Promise.all(
+      events.map(async event => {
+        const transactionId = event.id;
+        const exit = await exitHandler.methods
+          .exits(web3.utils.toHex(transactionId))
+          .call();
+
+        exits.push({ event, exit });
+
+        const exitSum = exits.reduce((acc, e) => {
+          return !e.exit.finalized ? new BN(e.exit.amount).add(acc) : acc;
+        }, new BN(0));
+
+        this.state.exitSum = parseFloat(web3.utils.fromWei(exitSum, "ether"));
+      })
+    );
+
+    logger.info(`PLASMA EXITS SUM: ${this.state.exitSum}`);
+  }
+
+  async getPlasmaBalance() {
+    const { plasma, web3, exitHandler, bridge } = leap;
+
+    const addresses = await plasma.getColors();
+    let plasmaBalance = 0;
+
+    await Promise.all(
+      addresses.map(async address => {
+        const contractInstance = await new web3.eth.Contract(ERC20Abi, address);
+        const exitHandlerBalance = await contractInstance.methods
+          .balanceOf(exitHandler.options.address)
+          .call();
+        const bridgeBalance = await contractInstance.methods
+          .balanceOf(bridge.options.address)
```

Comment:
why account for Bridge balance here? I don't think that Bridge contract ever holds any tokens. Am I missing something?

Reply:
my understanding is the SUM of the plasma chain should equal the UTXO + EXITSUM so need to be able to get that?